driver-mapping/src/main/java/com/datastax/driver/mapping/ReflectionMapper.java

```diff
@@ -162,6 +162,12 @@ public void setValue(Object entity, Object value) {
 
             super.setValue(entity, udtMapper.toEntity((udtValue)));
         }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        Object toSerializableValue(Object value) {
+            return value == null ? null : udtMapper.toUDT((U) value);
+        }
```

Comment:
We also need to override that in `NestedUDTMapper` and `EnumMapper` (using an enum as partition key, though unlikely, is possible).